lib/clientv2/workloadendpoint.go

```diff
@@ -90,3 +99,34 @@ func (r workloadEndpoints) List(ctx context.Context, opts options.ListOptions) (
 func (r workloadEndpoints) Watch(ctx context.Context, opts options.ListOptions) (watch.Interface, error) {
 	return r.client.resources.Watch(ctx, opts, apiv2.KindWorkloadEndpoint)
 }
+
+func (r workloadEndpoints) validate(res *apiv2.WorkloadEndpoint) error {
+	// Validate the workload endpoint indices and the name match.
+	wepids := names.WorkloadEndpointIdentifiers{
+		Node: res.Spec.Node,
+		Orchestrator: res.Spec.Orchestrator,
+		Endpoint: res.Spec.Endpoint,
+		Workload: res.Spec.Workload,
+		Pod: res.Spec.Pod,
+		ContainerID: res.Spec.ContainerID,
+	}
+	expectedName, err := wepids.CalculateWorkloadEndpointName(false)
+	if err != nil {
+		return err
+	}
+	if len(res.Name) == 0 {
```

Comment:
curious why can't we do `res.Name == ""`? That seems more readable

Reply:
i think it should have the same result but `len` is usually for a slice or array

Reply:
i think its what the k8s code does more often for empty string.
 should amount to the same thing

Reply:
Ok, not too fussed about it. Just looks a bit odd